server/src/main/java/org/elasticsearch/cluster/coordination/StableMasterHealthIndicatorService.java

```diff
@@ -260,6 +267,30 @@ private HealthIndicatorDetails getDetails(boolean explain, MasterHistory localMa
         };
     }
 
+    /**
+     * This returns HealthIndicatorDetails.EMPTY if explain is false, otherwise a HealthIndicatorDetails object containing only a
+     * "recent_masters" array and a "cluster_coordination" string. The "recent_masters" array will contain "recent_master" objects.
+     * Each "recent_master" object will have "node_id" and "name" fields for the master node. These fields will never be null because
+     * null masters are not written to this array. The "cluster_coordination" string will be the clusterCoordinationMessage passed in.
+     * @param explain If true, the HealthIndicatorDetails will contain "recent_masters" and "cluster_coordination". Otherwise it will be
+     *                empty.
+     * @param localMasterHistory The MasterHistory object to pull recent master info from
+     * @param clusterCoordinationMessage The cluster coordination message to put in the returned details
+     * @return An empty HealthIndicatorDetails if explain is false, otherwise a HealthIndicatorDetails containing only "recent_masters"
+     * and "cluster_coordination".
+     */
+    private HealthIndicatorDetails getDetails(boolean explain, MasterHistory localMasterHistory, String clusterCoordinationMessage) {
```

Comment:
Do we need a separate `getDetails` method or could we enhance the existing one with a `@Nullable clusterCoordinationMessage`? 

(or otherwise have 2 methods but reuse the `DETAILS_RECENT_MASTERS` and the null current master build-ups)

Reply:
I merged them into one.